vir/defs/high/ast/statement.rs

```diff
@@ -188,3 +189,11 @@ pub struct GhostAssignment {
     pub value: Expression,
     pub position: Position,
 }
+
+#[display(fmt = "{} := shorten_lifetime({:?}, {})", target, value, rd_perm)]
+pub struct ShortenLifetime {
+    pub target: VariableDecl,
+    pub value: Expression,
```

Comment:
Is this statement supposed to intersect a series of lifetimes and while doing so modify their `LifetimeTokens`? If yes, then using `Vec<LifetimeConst>` would make it easier to generate `vir_low` than `Expression`.

Reply:
I think you're right, this makes no sense. In vir_low I want to create something like this:

```
var old_lft6: Lifetime := lft6
lft6 := lft_tok_sep_take(bw0, bw1, q)
bor_shorten(lft6, old_lft6, q, _4.ref)
```

Having a vector would be better here. And `_4.ref` we will have to look up in vir_low, right?

Reply:
Does `ShortenLifetime` correspond to `lft_tok_sep_take`, to `bor_shorten`, or to both? If `bor_shorten` is needed, then I would say that you have to include `_4` into the struct.

Reply:
Originally I was intending to do `lft_tok_sep_take` only - but doing both at the same time seems nicer. I think some work will be required to include `_4`, not sure.

Reply:
renamed to `LifetimeTake`.